### PR TITLE
Adding Truncate nemesis

### PR DESCRIPTION
**Pull request by fruch**

### CODE REVIEW

File: sdcm/nemesis.py [hunk 1]
```diff
@@ -517,10 +518,37 @@ def disrupt_nodetool_cleanup(self):
...
+            data = session.execute(cql)
+            table_truncate_count = data[0].count
+        except InvalidRequest:
+            self.log.warning("Keyspace ks_truncate does not exist")
+        self.log.debug("fruch: table_truncate_count=%d", table_truncate_count)
+
+        # if key space doesn't exist or the table is empty, create it using c-s
+        if not (keyspace_truncate in test_keyspaces and table_truncate_count >= 1):
+            # create with stress tool
+            node.remoter.run('cassandra-stress write n=1000 cl=one -port jmx=6868 -mode native cql3 -schema keyspace="{}"'.format(keyspace_truncate), verbose=True, ignore_status=True)
```
Comment (roydahan) [inline sdcm/nemesis.py]: Change to n=400000

File: sdcm/nemesis.py [hunk 2]
```diff
@@ -517,10 +518,37 @@ def disrupt_nodetool_cleanup(self):
...
+        # get the count of the truncate table
+        test_keyspaces = self.cluster.get_test_keyspaces()
+        cql = "SELECT COUNT(*) FROM {}.{}".format(keyspace_truncate, table)
+        session = self.tester.cql_connection_patient(self.target_node)
+        try:
+            data = session.execute(cql)
+            table_truncate_count = data[0].count
+        except InvalidRequest:
+            self.log.warning("Keyspace ks_truncate does not exist")
+        self.log.debug("fruch: table_truncate_count=%d", table_truncate_count)
```
Comment (roydahan) [inline sdcm/nemesis.py]: remove the "fruch"

File: sdcm/nemesis.py (hunk 1 above)
Comment (roydahan) [reply sdcm/nemesis.py]: Change to cl=quorum

File: sdcm/nemesis.py [hunk 3]
```diff
@@ -517,10 +518,37 @@ def disrupt_nodetool_cleanup(self):
...
+        # do the actual truncation
+        self._set_current_disruption('TruncateMonkey {}'.format(node))
+        cql = 'TRUNCATE {}.{}'.format(keyspace_truncate, table)
+        self._run_in_cqlsh(cql)
+
+    def _run_in_cqlsh(self, cmd, node=None, **kwargs):
         if not node:
             node = self.target_node
-        node.remoter.run('cqlsh -e "{}" {}'.format(cmd, node.private_ip_address), verbose=True)
+        return node.remoter.run('cqlsh -e "{}" {}'.format(cmd, node.private_ip_address), verbose=True, **kwargs)
```
Comment (roydahan) [inline sdcm/nemesis.py]: not needed

File: sdcm/nemesis.py [hunk 4]
```diff
@@ -991,6 +1019,13 @@ def disrupt(self):
         self.disrupt_nodetool_cleanup()
 
 
+class TruncateMonkey(Nemesis):
+
+    @log_time_elapsed_and_status
+    def disrupt(self):
+        self.disrupt_truncate()
+
+
```
Comment (roydahan) [inline sdcm/nemesis.py]: Add also to the list of LimitedChaosMonkey

**Comment by roydahan** [comment]
@fruch reminding you to address the comments so we can merge it before we start the test of master.

**Comment by fruch** [comment]
@roydahan first thing tomorrow morning

File: sdcm/nemesis.py (hunk 4 above)
Comment (fruch) [reply sdcm/nemesis.py]: done

**Pull request merged**